### ISSUE TITLE
Training ControlNet + IP Adapter at the same time

### ISSUE THREAD

**Issue by Yunski** [tencent-ailab/IP-Adapter#324]
Do I need to set the attn processor when I train controlnet + ip adapter, like in the following code?

```
controlnet = ControlNetModel.from_pretrained(...)
controlnet.set_attn_processor(CNAttnProcessor())
controlnet.train()
```

**Comment by xiaohu2015** [comment]
you don't need this, you can just feed text features to controlnet.

**Comment by Yunski** [comment]
Thanks!

**Comment by flankechen** [comment]
@Yunski 
in the training code, you modify ipadapter forward with the sd pipeline like the inference .pynb code or just write the controlnet code directly?
in the original training code like:
noise_pred = ip_adapter(noisy_latents, timesteps, encoder_hidden_states, image_embeds)
 loss = F.mse_loss(noise_pred.float(), noise.float(), reduction="mean")
                
it would take some time to write the code in a pipeline style.
